File: Examples/VisualBasic/AsposePDF/Images/ConvertAllPagesToEMF.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Devices

Namespace AsposePDF.Images
    Public Class ConvertAllPagesToEMF
        Public Shared Sub Run()
            ' ExStart:ConvertAllPagesToEMF
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Images()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("ConvertAllPagesToEMF.pdf"))

            For pageCount As Integer = 1 To pdfDocument.Pages.Count
                Using imageStream As New FileStream("image" & pageCount & "_out" & ".emf", FileMode.Create)
                    ' Create Resolution object
                    Dim resolution As New Resolution(300)
                    ' Create EMF device with specified attributes
                    ' Width, Height, Resolution
                    Dim emfDevice As New EmfDevice(500, 700, resolution)

                    ' Convert a particular page and save the image to stream
                    emfDevice.Process(pdfDocument.Pages(pageCount), imageStream)

                    ' Close stream
                    imageStream.Close()

                End Using
            Next pageCount
            ' ExEnd:ConvertAllPagesToEMF
            System.Console.WriteLine("PDF pages are converted to EMF successfully!")
        End Sub
    End Class
End Namespace

File: Examples/CSharp/AsposePDF/Images/ConvertAllPagesToEMF.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Devices;

namespace CSharp.AsposePdf.Images
{
    public class ConvertAllPagesToEMF
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Images();

            //open document
            Document pdfDocument = new Document(dataDir+ "ConvertAllPagesToEMF.pdf");

            for (int pageCount = 1; pageCount <= pdfDocument.Pages.Count; pageCount++)
            {
                using (FileStream imageStream = new FileStream(dataDir + "image" + pageCount + ".emf", FileMode.Create))
                {
                    //create Resolution object
                    Resolution resolution = new Resolution(300);
                    //create PNG device with specified attributes
                    //Width, Height, Resolution
                    EmfDevice emfDevice = new EmfDevice(500, 700, resolution);

                    //convert a particular page and save the image to stream
                    emfDevice.Process(pdfDocument.Pages[pageCount], imageStream);

                    //close stream
                    imageStream.Close();

                }
            }

            
        }
    }
}